File: sort/go/src/sort/selectionSort.go
// sort/go/src/sort/selectionSort.go

package sort

// ------------------------------------------------------------------------
//     selectionSort  -   selection sort implementation
// ------------------------------------------------------------------------

// Selection sort function.
func SelectionSort(ar *[]float32, rev bool) {
	for i := range *ar {
		minIndex := i
		for j := i; j < len(*ar); j++ {
			if (!rev && (*ar)[j] < (*ar)[minIndex]) || (rev && (*ar)[j] > (*ar)[minIndex]) {
				minIndex = j
			}
		}
		if minIndex > i {
			swap(ar, i, minIndex)
		}
	}
}

File: sort/javascript/src/sort/selectionSort.js
// sort/javascript/src/sort/selectionSort.js

const swap = require("./swap").swap

// ------------------------------------------------------------------------
//     selectionSort  -   selection sort implementation
// ------------------------------------------------------------------------

// Selection sort function.
async function selectionSort(ar, rev=false) {
    for (let i = 0; i < ar.length; i++) {
        let minIndex = i;
        for (let j = i; j < ar.length; j++) {
            if ((!rev && ar[j] < ar[minIndex]) || (rev && ar[j] > ar[minIndex])) {
                minIndex = j
            }
        }
        if (minIndex > 1) {
            swap(ar, i ,minIndex)
        }
    }
}

exports.selectionSort = selectionSort;